src/Dialogs/Imap/ImapDialog.vala

```diff
@@ -253,21 +251,26 @@ public class OnlineAccounts.ImapDialog : Gtk.Window {
 
         login_page.cancel.connect (destroy);
 
-        login_page.next_button.clicked.connect (() => {
-            leaflet.visible_child = main_box;
+        main_page.shown.connect (() => {
             default_widget = save_button;
         });
 
+        login_page.next_button.clicked.connect (() => {
+            navigation_view.push (main_page);
+        });
+
+        login_page.shown.connect (() => {
+            default_widget = login_page.next_button;
+        });
+
         save_page.close.connect (destroy);
 
         save_page.back.connect (() => {
-            leaflet.navigate (Adw.NavigationDirection.BACK);
-            default_widget = save_button;
+            navigation_view.pop ();
         });
 
         back_button.clicked.connect (() => {
-            leaflet.navigate (Adw.NavigationDirection.BACK);
-            default_widget = login_page.next_button;
+            navigation_view.pop ();
```

Comment:
Another mem leak :(
(At least if back_buttons' life time depends on us, I'm not to deep in the codebase)